### ISSUE TITLE
[ENH] Nilearn plotting of RGB-type images

### ISSUE THREAD

**Issue by mattcieslak** [nilearn/nilearn#4224]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe your proposed enhancement in detail.

Opening this issue as a follow up to [this neurostars thread](https://neurostars.org/t/plotting-rgb-images-in-nilearn/27798/4)
There is also an existing issue that is somewhat similar: https://github.com/nilearn/nilearn/issues/522

## Background
In many dMRI methods we'll derive a direction encoded color (DEC) map that is used to show the principal direction of the WM in a voxel. The X Y and Z values of a unit vector get mapped to the R G and B channels respectively. Sometimes the A channel is weighted by FA or some other derived measure so the coloring is proportional to the certainty of the direction.

It also looks like some very early examples in this repo used RGB to encode how face/house/car-representing a voxel was. Perhaps since then the plotting methods made stronger assumptions about data being scalar? 

### Benefits to the change

Nilearn plotting is hands-down my favorite way to display volumetric imaging data, particularly when displaying images with multiple resolutions as foregrounds and backgrounds. I would love to be able to plot as T1w image as a background and a DEC map as an overlay.

Currently there is no python-based implementation of DEC plotting on a high-res background. The current standard practice is to resample the RGB image into the resolution of the background image and plot with imshow, selecting a slice based on the i,j,k of the high res image. This can introduce lots of orientation/resolution bugs.

### Pseudocode for the new behavior, if applicable

It would be awesome to have something like this 

```python
# Easiest version

plotting.view_img(
    color_fa_image,
    bg_img=background_t1w_image,
    title="Color FA",
    cmap="direct_rgba"
)
```

Maybe there's a way to do this using nilearn plotting as is? For example, maybe I could loop over the different color channels and add them as separate overlays?

There is some example data available from ITK Snap [here](http://www.nitrc.org/frs/download.php/6176/diffusion.zip) unfortunatrly `t1.nii.gz` isn't high res, but the `tensor_rgb.nii.gz` is a good example of what the RGB channels would look like and `tensor_fa.nii.gz` would be a good example of the alpha channel. @arokem probably has better examples of these kinds of images

**Comment by bthirion** [comment]
Thx for opening this ! 
I think I would prefer having a dedicated plotting function for multi-channel images (as `plot_anat`, `plot_epi`, `plot_stat_map`). This would minimize side effects etc. And we would need to demo this with a sample RGB image.

**Comment by mattcieslak** [comment]
[tensor_rgb.nii.gz](https://github.com/nilearn/nilearn/files/13957201/tensor_rgb.nii.gz)

This is a good example. It should look like this:
<img width="709" alt="image" src="https://github.com/nilearn/nilearn/assets/170026/f7276969-9958-43bd-8688-1f63d32c62c5">

**Comment by bthirion** [comment]
Is there a license associated with this image ?

**Comment by mattcieslak** [comment]
Yes, it's rather prohibitive. I will make some realistic phantom data that we can use/distribute freely

**Comment by arokem** [comment]
For a more realistic use-case, maybe we could use one of the HBN-POD2
subjects for this? I would be happy to calculate FA-DEC for one of those
subjects.

**Comment by mattcieslak** [comment]
Here is a link to better candidate data for testing: https://upenn.box.com/s/qz1ak694udeete9sfzez2hxvvf77a97x

It is from HBN-POD2 which has a CC 4.0 BY license based on this [info](http://fcon_1000.projects.nitrc.org/indi/cmi_healthy_brain_network/sharing_neuro.html)

**Comment by arokem** [comment]
There are plenty of openneuro datasets that could fit the bill, though! (right?)
